### PR TITLE
Add OpenEXR format support

### LINKED ISSUE #216: Support OpenEXR format for tensorflow-io
OpenEXR is an interesting format that could be very useful in computer vision. This issue is to track the support of OpenEXR for tensorflow-io.

### PR DESCRIPTION
**Pull request by yongtang**
This PR adds OpenEXR format support so that it is possible to extract channels from EXR formats.

Two APIs have been added, decode_exr_info and decode_exr.

This PR fixes #216

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow_io/core/kernels/image_openexr_kernels.cc [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+          dtype = DT_UINT32;
+          break;
+        case Imf::HALF:
+          dtype = DT_HALF;
+          break;
+        case Imf::FLOAT:
+          dtype = DT_FLOAT;
+          break;
+        default:
+          OP_REQUIRES(context, false, errors::InvalidArgument("invalid data type: ", pixel_type));
```
Comment (BryanCutler) [inline tensorflow_io/core/kernels/image_openexr_kernels.cc]: should this say "invalid pixel type"?

Comment (yongtang) [reply tensorflow_io/core/kernels/image_openexr_kernels.cc]: Thanks. That makes sense. The PR has been updated.

File: tensorflow_io/core/kernels/image_openexr_kernels.cc [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+  ~OpenEXRIStream() {}
+
+  bool read (char c[/*n*/], int n) override {
+    if (pos_ + n > size_) {
+      throw IEX_NAMESPACE::InputExc ("Unexpected end of file.");
+    }
+    StringPiece result;
+    Status status = file_->Read(pos_, n, &result, c);
+    if (!status.ok() || result.size() < n) {
+      throw IEX_NAMESPACE::InputExc ("Unexpected end of file.");
```
[the remark this answers is not in the document]
Comment (yongtang) [reply tensorflow_io/core/kernels/image_openexr_kernels.cc]: Updated.

File: tensorflow_io/core/kernels/image_openexr_kernels.cc [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+  : OPENEXR_IMF_INTERNAL_NAMESPACE::IStream(filename.c_str())
+  , file_(file)
+  , size_(size)
+  , pos_(0) {}
+
+  ~OpenEXRIStream() {}
+
+  bool read (char c[/*n*/], int n) override {
+    if (pos_ + n > size_) {
+      throw IEX_NAMESPACE::InputExc ("Unexpected end of file.");
```
[the remark this answers is not in the document]
Comment (yongtang) [reply tensorflow_io/core/kernels/image_openexr_kernels.cc]: Updated.

**Comment by yongtang** [comment]
Updated and all tests pass. Thanks for the review 👍

**Pull request merged**